gst/nnstreamer/tensor_common.c

```diff
@@ -979,45 +979,50 @@ gst_tensor_get_type (const gchar * typestr)
     return _NNS_END;
   }
 
-  if (type_string[0] == 'u' || type_string[0] == 'U') {
-    /**
-     * Let's believe the developer and the following three letters are "int"
-     * (case insensitive)
-     */
-    if (len == 6) {             /* uint16, uint32 */
-      if (type_string[4] == '1' && type_string[5] == '6')
+  if (g_regex_match_simple ("^uint(8|16|32|64)$",
+          type_string, G_REGEX_CASELESS, 0)) {
+    size = g_ascii_strtoull (&type_string[4], NULL, 10);
```

Comment:
function 'g_ascii_strtoull' returns the value type 'guint64'

Reply:
need cast for 32b/64b safety? (in stricter systems, compilers may incur warnings/errors?)